zulipterminal/helper.py

```diff
@@ -365,14 +368,21 @@ def classify_unread_counts(model: Any) -> UnreadCounts:
     )
 
     for pm in unread_msg_counts['pms']:
-        count = len(pm['unread_message_ids'])
+        msg_ids = pm['unread_message_ids']
+        count = len(msg_ids)
+        msg = {'type': 'pm', 'sender_id': pm['sender_id']}
+        unread_msgs.update(zip(msg_ids, [msg] * count))
```

Comment:
When updating counts of unread messages, we will need information about which stream/pm the message_id belongs to. 
Since the current index does not contain all the messages, simply storing message_ids will not be enough. Hence I've modified the index datastructure